weather_dl/download_pipeline/partition.py

```diff
@@ -193,13 +198,19 @@ def prepare_partitions_from_index(config: Config, indexes: t.List[Index]) -> t.I
         yield _create_partition_config(option, config)
 
 
-def new_downloads_only(candidate: Config, store: t.Optional[Store] = None) -> bool:
+def new_downloads_only(candidate: Config, store: t.Optional[Store] = None,
+                       manifest: Manifest = NoOpManifest(Location('noop://in-memory')),
+                       update_manifest: bool = False) -> bool:
     """Predicate function to skip already downloaded partitions."""
     if store is None:
         store = FSStore()
-    should_skip = skip_partition(candidate, store)
+    should_skip = skip_partition(candidate, store, manifest)
```

Comment:
I'm concerned about having a side effect in a `Filter` operation. I'd rather let this be a predicate, and include the update logic in a separate beam step later on.

Reply:
Updated the code.